src/object.hpp

```diff
@@ -0,0 +1,91 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_OBJECT_HPP
+#define REALM_OS_OBJECT_HPP
+
+#include "shared_realm.hpp"
+
+#include <realm/row.hpp>
+
+namespace realm {
+class CollectionChangeCallback;
+class ObjectSchema;
+struct NotificationToken;
+
+namespace _impl {
+    class ObjectNotifier;
+}
+
+class Object {
+public:
+    Object(SharedRealm r, ObjectSchema const& s, BasicRowExpr<Table> const& o)
+    : m_realm(std::move(r)), m_object_schema(&s), m_row(o) { }
+
+    Object(SharedRealm r, ObjectSchema const& s, Row const& o)
+    : m_realm(std::move(r)), m_object_schema(&s), m_row(o) { }
+
+    ~Object();
+
+    // property getter/setter
+    template<typename ValueType, typename ContextType>
+    void set_property_value(ContextType ctx, std::string prop_name,
```

Comment:
It's a little weird that these methods are declared in this class but the implementation lives over in object_accessor.hpp.